pkg/policy/rule.go

```diff
@@ -37,29 +37,21 @@ type ruleMetadata struct {
 	// mutex protects all fields in this type.
 	Mutex lock.RWMutex
 
-	// localRuleConsumers is the set of the endpoints which this rule selects
-	// which are node-local.
-	EndpointsSelected map[uint16]*identity.Identity
-
-	// AllEndpoints tracks which node-local Endpoints have been 'processed'.
-	// That is, it determines whether the Endpoint has actually been processed
-	// in relation to this rule. It does *not* encode whether the rule selects
-	// the endpoint; that is what EndpointsSelected is for.
-	AllEndpoints map[uint16]struct{}
+	// IdentitySelected is a cache that maps from an identity to whether
+	// this rule selects that identity.
+	IdentitySelected map[*identity.Identity]bool
 }
 
 func newRuleMetadata() *ruleMetadata {
 	return &ruleMetadata{
-		EndpointsSelected: make(map[uint16]*identity.Identity),
-		AllEndpoints:      make(map[uint16]struct{}),
+		IdentitySelected: make(map[*identity.Identity]bool),
```

Comment:
Does it matter that two Identity instances could describe the same security identity? If yes then maybe use the numeric identity as the map key instead?

Reply:
It doesn't matter, this is purely a cache and if an entry is not found, it will be populated.

Do you know if this happens?